node: Provide context in log when restoring router addresses

tgraf opened the pull request:
The following log messages lacked context on how the address has been restored:

    cilium-agent[32467]: level=info msg="Restored IPv6 router IP: fc00::10ca:1" subsys=node

Provide the node_config file or netdevice context when restoring router addresses.

Fixes: #6776

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9947)
<!-- Reviewable:end -->

joestringer commented on pkg/node/node_address.go:
```diff
@@ -437,6 +437,11 @@ func getCiliumHostIPs() (ipv4GW, ipv6Router net.IP) {
 	if ipv4GW != nil {
+		log.WithFields(logrus.Fields{
+			"ipv4": ipv4GW,
+			"ipv6": ipv6Router,
```
Can `ipv4GW` be nil but `ipv6Router` non-nil here, leading to no message being printed?

tgraf replied:
Good point. Fixed

tgraf committed:
node: Provide context in log when restoring router addresses

The following log messages lacked context on how the address has been restored:

    cilium-agent[32467]: level=info msg="Restored IPv6 router IP: fc00::10ca:1" subsys=node

Provide the node_config file or netdevice context when restoring router addresses.

Fixes: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>
